examples/AlwaysSucceeds.purs

```diff
@@ -62,13 +60,24 @@ example cfg = launchAff_ do
 
 payToAlwaysSucceeds :: ValidatorHash -> Contract () TransactionHash
 payToAlwaysSucceeds vhash = do
+  -- Send to own stake credential. This is used to test mustPayToScriptAddress.
+  mbStakeKeyHash <- ownStakePubKeyHash
   let
     constraints :: TxConstraints Unit Unit
     constraints =
-      Constraints.mustPayToScript vhash unitDatum
-        Constraints.DatumWitness
-        $ Value.lovelaceValueOf
-        $ BigInt.fromInt 2_000_000
+      case mbStakeKeyHash of
+        Nothing ->
+          Constraints.mustPayToScript vhash unitDatum
+            Constraints.DatumWitness
+            $ Value.lovelaceValueOf
+            $ BigInt.fromInt 2_000_000
+        Just stakeKeyHash ->
+          Constraints.mustPayToScriptAddress vhash
+            (PubKeyCredential $ unwrap stakeKeyHash)
+            unitDatum
+            Constraints.DatumWitness
+            $ Value.lovelaceValueOf
+            $ BigInt.fromInt 2_000_000
```

Comment:
I think mine is more readable, because the cases are separated more clearly

Reply:
The problem with easy-purescript-nix upgrade is not so trivial, we seemingly can't just revert it, because we have `nixpkgs.follows = "ogmios/nixpkgs";`, and that version does not have `nodejs_12x`...

Reply:
This seems to work, we'll need to add a TODO issue here to drop this overlay once it's fixed in later versions (we should tell spago)
```diff
diff --git a/flake.nix b/flake.nix
index 26da7e38f..b9b37348f 100644
--- a/flake.nix
+++ b/flake.nix
@@ -220,6 +220,22 @@
           easy-ps = import inputs.easy-purescript-nix { pkgs = final; };
           purescriptProject = import ./nix { pkgs = final; };
         };
+        spago = final: prev: {
+          easy-ps = prev.easy-ps // {
+            spago = prev.easy-ps.spago.overrideAttrs (_: rec {
+              version = "0.20.7";
+              src = if final.stdenv.isDarwin
+                then final.fetchurl {
+                  url = "https://github.com/purescript/spago/releases/download/${version}/macOS.tar.gz";
+                  sha256 = "0s5zgz4kqglsavyh7h70zmn16vayg30alp42w3nx0zwaqkp79xla";
+                }
+                else final.fetchurl {
+                  url = "https://github.com/purescript/spago/releases/download/${version}/Linux.tar.gz";
+                  sha256 = "0bh15dr1fg306kifqipnakv3rxab7hjfpcfzabw7vmg0gsfx8xka";
+                };
+            });
+          };
+        };
         # This is separate from the `runtime` overlay below because it is
         # optional (it's only required if using CTL's `applyArgs` effect).
         # Including it by default in the `overlays.runtime` also requires that
```